Topics/Software_Tools/vim/vim.md

```diff
@@ -0,0 +1,152 @@
+# Introduction to Vim
+
+## Introduction
+
+
+### What is Vim?
+
+Vim, short for Vi Improved, is a light-weight, customizable and powerful text
+editor that comes with most UNIX-like operating systems (Linux, BSD, macOS).
+What separates Vim from other text editors and IDEs is it's unique methods of
+traversing text files and making edits, all without the need for using your
+mouse.  In addition, Vim runs entirely within the terminal, meaning it has a
+smaller memory footprint and generally runs faster than something like VSCode.
+
+### Why Vim?
+
+One of the strongest arguments against using Vim over something like VSCode is
+that Vim has a steep learning curve. While it is true that Vim has a steeper
+learning curve, learning the basics of navigation and editing with Vim is not a
+complicated task. Furthermore, knowing how to use Vim can be a useful skill on
+the job. For instance, when working with remote machines, you will often need to
+edit things like system configuration files, systemd services etc. Furthermore, you'll
+often have to SSH into them from the terminal, meaning you won't have access to other
+editors, only Vim.
+
+
+### Installation
+
+#### Linux
+
+Run the following command(s) in the terminal (specific to the Linux distribution you have installed)
+
+##### Debian/Ubuntu
+
+```
+sudo apt update
+sudo apt install vim
+```
+
+##### Red Hat/CentOS
+
+```
+sudo yum install vim-enhanced
+```
+
+##### Fedora
+
+```
+sudo dnf install vim
+```
+
+#### macOS
+
+Using Homebrew:
+
+If you have Homebrew installed:
+
+```
+brew install vim
+```
+
+Using MacPorts:
+
+If you have MacPorts installed:
+
+```
+sudo port install vim
+```
+
+#### Windows
+
+Using Chocolatey (Package Manager for Windows):
+
+If you have Chocolatey installed:
+
+```
+choco install vim
+```
+
+## Basics
+
+Manual Installation:
+
+- Go to the Vim website: https://www.vim.org/download.php
+
+- Download the latest version for Windows
+
+- Run the installer and follow the installation prompts
+
+
+### Modes
+Here is a diagram of the modes available on vim and how to access them, they will be used in the following sections.
+![Image](./vim_modes.png)
+
```

Comment:
missing reference to image -> please specify that this image is taken from the below link